src/cdi_analytic/Analytic_CP_Eloss.hh

```diff
@@ -34,25 +35,22 @@ public:
   typedef std::shared_ptr<Analytic_Eloss_Model> SP_Analytic_Model;
   typedef std::vector<double> sf_double;
   typedef std::string std_string;
+  typedef rtt_cdi::CParticle CParticle;
+  typedef rtt_cdi::CPModelAngleCutoff CPModelAngleCutoff;
 
 private:
   // Analytic eloss model.
   SP_Analytic_Model analytic_model;
 
-  // Particle type being transported.
-  int32_t projectile_zaid;
-
-  // Target species.
-  int32_t target_zaid;
-
   // Data model (e.g. Eloss, large-angle scatter, etc.)
   // Currently only ELOSS is implemented.
   //rtt_cdi::CPModel cpmodel;
 
 public:
   // Constructor.
-  Analytic_CP_Eloss(SP_Analytic_Model model_in, int32_t target_zaid_in,
-                    int32_t projectile_zaid_in);
+  Analytic_CP_Eloss(SP_Analytic_Model model_in, CParticle target_in,
+                    CParticle projectile_in,
+                    CPModelAngleCutoff model_angle_cutoff_in);
 
   // >>> ACCESSORS
   const_SP_Model get_Analytic_Model() const { return analytic_model; }
```

Comment:
This line isn't covered by your unit test.  Please add a test.

Reply:
Done.